src/main/python/org/broadinstitute/hellbender/tools/gcnv/gcnvkernel/models/model_ploidy.py

```diff
@@ -0,0 +1,364 @@
+                    contig_ploidy_prior_map[contig][ploidy] = values[j]
+        return contig_ploidy_prior_map
+
+    @staticmethod
+    def expose_args(args: argparse.ArgumentParser, hide: Set[str] = None):
+        """Exposes arguments of `__init__` to a given instance of `ArgumentParser`.
+
+        Args:
+            args: an instance of `ArgumentParser`
+            hide: a set of arguments not to expose
+
+        Returns:
+            None
+        """
+        group = args.add_argument_group(title="Copy number calling parameters")
+        if hide is None:
+            hide = set()
+
+        initializer_params = inspect.signature(PloidyModelConfig.__init__).parameters
+        valid_args = {"--" + arg for arg in initializer_params.keys()}
+        for hidden_arg in hide:
+            assert hidden_arg in valid_args, \
+                "Initializer argument to be hidden {0} is not a valid initializer arguments; possible " \
+                "choices are: {1}".format(hidden_arg, valid_args)
+
+        def process_and_maybe_add(arg, **kwargs):
+            full_arg = "--" + arg
+            if full_arg in hide:
+                return
+            kwargs['default'] = initializer_params[arg].default
+            group.add_argument(full_arg, **kwargs)
+
+        process_and_maybe_add("mean_bias_sd",
+                              type=float,
+                              help="Contig-level mean coverage bias standard deviation",
+                              default=initializer_params['mean_bias_sd'].default)
+
+        process_and_maybe_add("mapping_error_rate",
+                              type=float,
+                              help="Typical mapping error rate",
+                              default=initializer_params['mapping_error_rate'].default)
+
+        process_and_maybe_add("psi_j_scale",
+                              type=float,
+                              help="Typical scale of contig-specific unexplained coverage variance",
+                              default=initializer_params['psi_j_scale'].default)
+
+        process_and_maybe_add("psi_s_scale",
+                              type=float,
+                              help="Typical scale of sample-specific unexplained coverage variance",
+                              default=initializer_params['psi_s_scale'].default)
+
+    @staticmethod
+    def from_args_dict(args_dict: Dict):
+        """Initialize an instance of `PloidyModelConfig` from a dictionary of arguments.
+
+        Args:
+            args_dict: a dictionary of arguments; the keys must match argument names in
+                `PloidyModelConfig.__init__`
+
+        Returns:
+            an instance of `PloidyModelConfig`
+        """
+        relevant_keys = set(inspect.getfullargspec(PloidyModelConfig.__init__).args)
+        relevant_kwargs = {k: v for k, v in args_dict.items() if k in relevant_keys}
+        return PloidyModelConfig(**relevant_kwargs)
+
+
+class PloidyWorkspace:
```

Comment:
Just a thought: if we split off the inference package, it would be useful to have a simple model as a pedagogical example for developers.  Such a model could highlight tricks like the sharing of tensors here.

Reply:
I completely agree, and I think it is a good idea to do so. Will make an issue.

Reply:
Oh nevermind, you already made it.